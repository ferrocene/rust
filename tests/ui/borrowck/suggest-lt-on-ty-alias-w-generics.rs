//@ revisions: preedition2021 postedition2021
//@[preedition2021] edition: ..2018
//@[postedition2021] edition: 2021..

type Lazy<T> = Box<dyn Fn() -> T + 'static>;

fn test(x: &i32) -> Lazy<i32> {
    Box::new(|| {
        //~^ ERROR lifetime may not live long enough
        //[preedition2021]~| ERROR closure may outlive the current function
        *x
    })
}

fn main() {}
