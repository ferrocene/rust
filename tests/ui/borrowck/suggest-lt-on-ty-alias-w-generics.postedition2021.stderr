error: lifetime may not live long enough
  --> $DIR/suggest-lt-on-ty-alias-w-generics.rs:8:5
   |
LL |   fn test(x: &i32) -> Lazy<i32> {
   |              - let's call the lifetime of this reference `'1`
LL | /     Box::new(|| {
LL | |
LL | |
LL | |         *x
LL | |     })
   | |______^ returning this value requires that `'1` must outlive `'static`
   |
help: to declare that the trait object captures data from argument `x`, you can add a lifetime parameter `'a` in the type alias
   |
LL - type Lazy<T> = Box<dyn Fn() -> T + 'static>;
LL + type Lazy<'a, T> = Box<dyn Fn() -> T + 'a>;
   |

error: aborting due to 1 previous error

